### ISSUE TITLE
Adding manual commission (FM) with PNR_AddMultiElements

### ISSUE THREAD

**Issue by fayax** [amabnl/amadeus-ws-client#45]
Hello,

How do I add FM with PNR_AddMultiElements to PNR?
I think it is already implemented. Any example will be appreciated.

Thanks

**Comment by DerMika** [comment]
Hi, 

_(Besides using the message FOP_CreateFormOfPayment)_ there are 2 samples to add an FP to the PNR with PNR_AddMultiElements here: https://github.com/amabnl/amadeus-ws-client/blob/master/docs/samples/pnr-create-modify.rst#form-of-payment-fp

They are an `FP CASH` and an `FP CC`sample. To know how to enter the element you need, I need to know exactly what you want to do. The title of your issue seems to be contradicting itself, because a Manual Commission entry is an `FM` element, not an `FP` element.

The library doesn't support FM elements just yet, but I can add it.

**Comment by fayax** [comment]
Sorry, typo. I mean FM.
Appreciate if you can add support for FM.

**Comment by DerMika** [comment]
Can you please check the [functional docs](https://webservices.amadeus.com/extranet/kdbViewDocument.do?externalId=HTML_UG_WBS_PNR_AddMultiElements_PNRADD_16.1&docStatus=Published&mpId=se__25_1__8_functional&serviceVersionId=4658) of `Fare_MasterPricerTravelBoardSearch` and tell me if you need anything more beyond the sample given in **5.28 Operation: Manual Commission**, or if it is good enough for you if I implement support for that sample?

**Comment by fayax** [comment]
I think that example is enough. Amadeus support provided me this sample code.

```
<elementManagementData>
<reference>
<qualifier>OT</qualifier>
<number>2</number>
</reference>
<segmentName>FM</segmentName>
</elementManagementData>
<commission>
<passengerType>PAX</passengerType>
<indicator>FM</indicator>
<commissionInfo>
<percentage>0</percentage>
</commissionInfo>
</commission>
</dataElementsIndiv>

```

**Comment by DerMika** [comment]
Support for `FM` elements is now added. 

Check the docs for a sample: https://github.com/amabnl/amadeus-ws-client/blob/master/docs/samples/pnr-create-modify.rst#manual-commission-fm

Don't hesitate to reopen this issue if it's not working for you!

**Comment by fayax** [comment]
Thanks for the quick implementation.
MPTBS is not working for me now.
I get this error message.

`      <Fare_MasterPricerTravelBoardSearchReply xmlns="http://xml.amadeus.com/FMPTBR_14_3_1A">
         <errorMessage>
            <applicationError>
               <applicationErrorDetail>
                  <error>118</error>
               </applicationErrorDetail>
            </applicationError>
            <errorMessageText>
               <freeTextQualification>
                  <textSubjectQualifier>3</textSubjectQualifier>
               </freeTextQualification>
               <description>Transaction unable to process : AVL (712 / 301)</description>
            </errorMessageText>
         </errorMessage>
      </Fare_MasterPricerTravelBoardSearchReply>`

I'm not sure if it is related with the latest commit. I was at v1.1.1 and then upgraded to 1.2.1 with FM implementation.  Kindly advice.

**Comment by DerMika** [comment]
I've just tested 1.2.1 with a few MasterPricer calls, and they work fine here. Perhaps a temporary issue?

So... I can't reproduce this with the information I have now. Can you open a separate issue with the MasterPricer call you're trying to perform so I can try to reproduce?

**Comment by fayax** [comment]
It seems that it is a temporary issue. Works fine now. Thank you!

**Comment by DerMika** [comment]
Also, FM element support is not included in 1.2.1, it will be in 1.3.0. you'll need to switch your composer to 'dev-master' to test it.

**Comment by AhmedAbdallahNouh** [comment]
@fayax  PLAEASE can you help me with this : i having this response : 

<s:Body>
    <Fare_MasterPricerTravelBoardSearchReply xmlns="http://xml.amadeus.com/FMPTBR_18_1_1A">
      <errorMessage>
        <applicationError>
          <applicationErrorDetail>
            <error>978</error>
          </applicationErrorDetail>
        </applicationError>
        <errorMessageText>
          <freeTextQualification>
            <textSubjectQualifier>1</textSubjectQualifier>
          </freeTextQualification>
          <description>Invalid option</description>
        </errorMessageText>
      </errorMessage>
    </Fare_MasterPricerTravelBoardSearchReply>
  </s:Body>   

when im trying to search with business cabin class 

 <s:Body xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
    <Fare_MasterPricerTravelBoardSearch xmlns="http://xml.amadeus.com/FMPTBQ_18_1_1A">
      <numberOfUnit>
        <unitNumberDetail>
          <numberOfUnits>1</numberOfUnits>
          <typeOfUnit>PX</typeOfUnit>
        </unitNumberDetail>
        <unitNumberDetail>
          <numberOfUnits>250</numberOfUnits>
          <typeOfUnit>RC</typeOfUnit>
        </unitNumberDetail>
      </numberOfUnit>
      <paxReference>
        <ptc>ADT</ptc>
        <traveller>
          <ref>1</ref>
        </traveller>
      </paxReference>
      <fareFamilies>
        <familyInformation>
          <fareFamilyname>FFAMILY1</fareFamilyname>
          <hierarchy>80</hierarchy>
        </familyInformation>
        <familyCriteria>
          <corporateInfo>
            <corporateName>NONCORP</corporateName>
          </corporateInfo>
          <cabinProduct>
            <cabinDesignator>Y</cabinDesignator>
          </cabinProduct>
        </familyCriteria>
      </fareFamilies>
      <fareOptions>
        <pricingTickInfo>
          <pricingTicketing>
            <priceType>ET</priceType>
            <priceType>RP</priceType>
            <priceType>RU</priceType>
          </pricingTicketing>
        </pricingTickInfo>
      </fareOptions>
      <itinerary>
        <requestedSegmentRef>
          <segRef>1</segRef>
        </requestedSegmentRef>
        <departureLocalization>
          <departurePoint>
            <locationId>NCE</locationId>
          </departurePoint>
        </departureLocalization>
        <arrivalLocalization>
          <arrivalPointDetails>
            <locationId>AMS</locationId>
          </arrivalPointDetails>
        </arrivalLocalization>
        <timeDetails>
          <firstDateTimeDetail>
            <date>250724</date>
          </firstDateTimeDetail>
        </timeDetails>
      </itinerary>
    </Fare_MasterPricerTravelBoardSearch>
  </s:Body>
